src/Tribe/Editor/Compatibility.php

```diff
@@ -17,6 +30,164 @@ class Tribe__Events__Editor__Compatibility {
 	 */
 	public function hook() {
 		add_action( 'admin_init', array( $this, 'deactivate_gutenberg_extension_plugin' ) );
+		add_action( 'add_meta_boxes', array( $this, 'maybe_add_purge_blocks_metabox' ) );
+		tribe_notice(
+			'events-editor-compatibility-purge-blocks',
+			array( $this, 'purge_blocks_notice' ),
+			array(
+				'type'    => 'success',
+				'dismiss' => 1,
+				'wrap'    => 'p',
+			),
+			array( $this, 'should_display_purge_blocks_notice' )
+		);
+	}
+
+	/**
+	 * What is the notice when we purge blocks
+	 *
+	 * @since  TBD
+	 *
+	 * @return string
+	 */
+	public function purge_blocks_notice() {
+		return esc_html__( 'Clean up complete. This Event is now optimized for the classic editor.', 'the-events-calendar' );
+	}
+
+	/**
+	 * Check if we need to display the warning that we updated the Blocks to classic
+	 *
+	 * @since  TBD
+	 *
+	 * @return bool
+	 */
+	public function should_display_purge_blocks_notice() {
+		$action = tribe_get_request_var( 'tribe-action', false );
+
+		// Bail if incorrect action
+		if ( 'purge-event-blocks' !== $action ) {
+			return false;
+		}
+		$post = absint( tribe_get_request_var( 'post' ) );
+
+		return $this->purge_blocks( $post );
+	}
+
+	/**
+	 * Do the actual purgin  of data.
+	 *
+	 * @since  TBD
+	 *
+	 * @param int $post Which post we will purge
+	 *
+	 * @return bool
+	 */
+	public function purge_blocks( $post ) {
+		// Bail on empty post
+		if ( empty( $post ) ) {
+			return false;
+		}
+
+		// Bail on non numeric post
+		if ( ! is_numeric( $post ) ) {
+			return false;
+		}
+
+		/** @var Tribe__Editor $editor */
+		$editor = tribe( 'editor' );
+
+		// Bail if not an event (might need to be removed later)
+		if ( ! tribe_is_event( $post ) ) {
+			return false;
+		}
+
+		// Bail if it doesn't have Blocks
+		if ( ! has_blocks( $post ) ) {
+			return false;
+		}
+
+		// Fetch WP_Post
+		$post = get_post( $post );
+
+		$post_content = preg_replace( '/\<\!\-\- \/?wp\:.*\/?-->/i', '', $post->post_content );
```

Comment:
Should we namespace to remove only blocks created by `tribe` otherwise we are removing all blocks that might be added into the event from other sources like WP or external plugins, unless thats the expected behavior.